src/stores/orderStore.ts

```diff
@@ -145,33 +148,37 @@ class OrderStore extends BaseStore {
   };
 
   private orderPlacedUnsuccessfully = (error: any) => {
-    console.error(error);
-
-    const needConfirmation =
-      error.status === 400 &&
-      error.message &&
-      JSON.parse(error.message).Confirmation;
-
-    if (needConfirmation) {
-      this.modalStore.addModal(
-        ModalMessages.expired,
-        null,
-        null,
-        Types.Expired
-      );
-      return;
+    const messageObject = JSON.parse(error.message);
+    if (messageObject) {
+      const key = Object.keys(messageObject)[0];
+
+      if (error.status === 400) {
+        switch (key) {
+          case 'Confirmation': {
+            this.modalStore.addModal(
+              ModalMessages.expired,
+              null,
+              null,
+              Types.Expired
+            );
+            return;
+          }
+          case 'AssetKycNeeded': {
+            this.modalStore.addModal(null, null, null, Types.MissedKyc);
+            return;
+          }
+          default:
+            {
+              const message = messageObject[key];
+              this.notificationStore.addNotification(
+                levels.error,
+                `${message}`
+              );
+            }
+            break;
+        }
+      }
     }
-
-    let message;
-    try {
-      message = JSON.parse(error.message).ME || JSON.parse(error.message);
-      message = getRestErrorMessage(message);
-    } catch (e) {
-      message = !!error.message.length ? error.message : messages.defaultError;
-    }
-    this.notificationStore.addNotification(levels.error, `${message}`);
-
-    return Promise.reject(error);
```

Comment:
it should return Promise.reject for right catching in the Order component. See Order.tsx, "applyOrder" - component